tests/unit/customizations/eks/test_get_token.py

```diff
@@ -88,13 +96,16 @@ def test_token(self):
 
     @patch.object(TokenGenerator, '_get_presigned_url', return_value='aHR0cHM6Ly9zdHMuYW1hem9uYXdzLmNvbS8=')
     def test_token_no_padding(self, mock_presigned_url):
-        generator = TokenGenerator(REGION, self._session_handler)
+        generator = TokenGenerator(REGION, self._session)
         tok = generator.get_token(CLUSTER_NAME, None)
         self.assertTrue('=' not in tok)
 
-    def test_token_sess(self):
-        generator = TokenGenerator(REGION, self._assuming_handler)
-        token = generator.get_token(CLUSTER_NAME, "RoleArn")
+    @patch.object(botocore.session.Session, 'create_client')
```

Comment:
Rather than patching session, you should inject a mock session that returns the client you want.

Reply:
Is it ok to keep this patch? I'd have to inject a lot of mocks to not only handle the `create_client` case, but also mock all of the request signing required types, and I haven't figured out all of the numerous mocks that would be needed to be created and passed through. (`botocore.credentials.Credentials`, `session.get_component('event_emitter')`, etc)

Reply:
I think this is difficult to patch as a consequence of the `SessionHandler` being a bit redundant now that we're using the default session provided by the CLI.

 Right now it's being used for two purposes:
1) Access to an `sts` client to make the assume role call
2) A vessel to carry the credentials object retrieved from assume role.

Furthering this, it seems a little strange that we would need to use the `RequestSigner` class directly (thus exposing more innards of the session / increasing complexity & testing burden).

From what I can tell it looks like we're doing this because we need to inject a custom header? If this is the only reason, simply registering a customization handler to the `sts` client on `before-sign` that injects this header seems to be a much more direct approach as it would allow us to just use the `generate_presigned_url` included in boto3.

I think there's quite a bit of refactoring / simplification we can do on this customization, though I'm leaning on the side of that being a follow up PR and not blocking this one.